extensions/mssql/src/objectExplorerNodeProvider/hdfsProvider.ts

```diff
@@ -51,7 +52,7 @@ export abstract class HdfsFileSourceNode extends TreeNode {
 	}
 
 	public async delete(recursive: boolean = false): Promise<void> {
-		await this.fileSource.delete(this.hdfsPath, recursive);
+		await (await this.getFileSource()).delete(this.hdfsPath, recursive);
```

Comment:
Can you move this inner await to the line above this one? Having nested awaits can be confusing at a glance.

Reply:
Yeah sure, I'll do that for all the nested awaits.